go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -114,44 +102,47 @@ func (npw *nodePortWatcher) AddService(service *kapi.Service) {
 						npw.gwBridge, stderr, err, flow)
 				}
 			}
-		}
-		for _, externalIP := range service.Spec.ExternalIPs {
-			if err := util.ValidatePort(svcPort.Protocol, svcPort.Port); err != nil {
-				klog.Errorf("Skipping service add for svc: %s, err: %v", svcPort.Name, err)
-				continue
-			}
-			flowProtocol := protocol
-			nw_dst := "nw_dst"
-			if utilnet.IsIPv6String(externalIP) {
-				flowProtocol = protocol6
-				nw_dst = "ipv6_dst"
-			}
-			_, stderr, err := util.RunOVSOfctl("add-flow", npw.gwBridge,
-				fmt.Sprintf("priority=100, in_port=%s, %s, %s=%s, tp_dst=%d, actions=%s",
-					npw.ofportPhys, flowProtocol, nw_dst, externalIP, svcPort.Port, npw.ofportPatch))
-			if err != nil {
-				klog.Errorf("Failed to add openflow flow on %s for ExternalIP: "+
-					"%s, stderr: %q, error: %v", npw.gwBridge, externalIP, stderr, err)
+
+			for _, externalIP := range service.Spec.ExternalIPs {
+				if err := util.ValidatePort(svcPort.Protocol, svcPort.Port); err != nil {
+					klog.Errorf("Skipping service add for svc: %s, err: %v", svcPort.Name, err)
+					continue
+				}
```

Comment:
this is not needed, the Service object is already validated by the Kubernetes API

Reply:
will remove